src/test/java/plugins/WorkflowPluginTest.java

```diff
@@ -196,30 +201,49 @@
 
     @Category(DockerTest.class)
     @WithDocker
-    @WithPlugins({"workflow-aggregator@1.14", "docker-workflow@1.4", "git", "ssh-agent@1.10"})
+    @WithPlugins({"workflow-cps@2.0", "workflow-job@2.0", "workflow-durable-task-step@2.0", "docker-workflow@1.5", "git", "ssh-agent@1.10", "ssh-slaves@1.11"})
     @WithCredentials(credentialType=WithCredentials.SSH_USERNAME_PRIVATE_KEY, values={"git", "/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/unsafe"}, id="gitcreds")
     @Issue("JENKINS-27152")
     @Test public void sshGitInsideDocker() throws Exception {
-        // Pending https://github.com/jenkinsci/docker-workflow-plugin/pull/31 (and additional APIs in DockerFixture etc.)
-        // it is not possible to run this on a bind-mounted Docker slave, so we cannot verify that paths are local to the slave.
-        GitContainer container = gitServer.get();
-        String host = container.host();
-        int port = container.port();
+        GitContainer gitContainer = gitServer.get();
         GitRepo repo = new GitRepo();
         repo.commit("Initial commit");
-        repo.transferToDockerContainer(host, port);
+        repo.transferToDockerContainer(gitContainer.host(), gitContainer.port());
+        DumbSlave slave = jenkins.slaves.create(DumbSlave.class);
+        slave.setExecutors(1);
+        slave.remoteFS.set("/home/test"); // TODO perhaps should be a constant in SshdContainer
+        SshSlaveLauncher launcher = slave.setLauncher(SshSlaveLauncher.class);
+        Process proc = new ProcessBuilder("stat", "-c", "%g", "/var/run/docker.sock").start();
+        String group = IOUtils.toString(proc.getInputStream()).trim();
+        Assume.assumeThat("docker.sock can be statted", proc.waitFor(), is(0));
+        try {
+            Integer.parseInt(group);
+        } catch (NumberFormatException x) {
+            Assume.assumeNoException("unexpected output from stat on docker.sock", x);
+        }
+        // Note that we need to link to the Git container both on the agent, for the benefit of JGit (TODO pending JENKINS-30600), and in the build container, for the benefit of git-pull:
```

Comment:
If you disable the JENKINS-30600 workarounds, you get the expected

```
Started by user anonymous
[Pipeline] node
Running on … in /home/test/workspace/…
[Pipeline] {
[Pipeline] sh
[…] Running shell script
+ docker inspect -f . cloudbees/java-build-tools
.
[Pipeline] withDockerContainer
$ docker run -t -d -u … --link=…:git -w /home/test/workspace/… --volumes-from … -e … --entrypoint cat cloudbees/java-build-tools
[Pipeline] {
[Pipeline] git
Cloning the remote Git repository
Cloning repository ssh://git@git/home/git/gitRepo
 > git init /home/test/workspace/… # timeout=10
ERROR: Error cloning remote repo 'origin'
hudson.plugins.git.GitException: Could not init /home/test/workspace/…
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$5.execute(CliGitAPIImpl.java:663)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$2.execute(CliGitAPIImpl.java:464)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:152)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:145)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at …
	at ......remote call to …(Native Method)
	at …
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:145)
	at …
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:131)
	at com.sun.proxy.$Proxy79.execute(Unknown Source)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1057)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1097)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:109)
	at …
Caused by: hudson.plugins.git.GitException: Error performing command: git init /home/test/workspace/…
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1791)
	at …
Caused by: java.io.IOException: Cannot run program "git" (in directory "/home/test/workspace/…"): error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at hudson.Proc$LocalProc.<init>(Proc.java:245)
	at hudson.Proc$LocalProc.<init>(Proc.java:214)
	at hudson.Launcher$LocalLauncher.launch(Launcher.java:846)
	at hudson.Launcher$ProcStarter.start(Launcher.java:384)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1780)
	... 15 more
Caused by: java.io.IOException: error=2, No such file or directory
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 20 more
[Pipeline] }
$ docker stop --time=1 …
$ docker rm -f …
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: null
Finished: FAILURE
```

and thus might serve as an integration test for https://github.com/jenkinsci/git-client-plugin/pull/219 (CC @ndeloof).